recipes/Switchboard/ASR/transformer/train.py

```diff
@@ -0,0 +1,545 @@
+
+        # Select a speech segment from the sph file
+        # start and end times are already frames.
+        # This is done in data preparation stage.
+        start = int(start)
+        stop = int(stop)
+        num_frames = stop - start
+        sig, fs = torchaudio.load(
+            wav, num_frames=num_frames, frame_offset=start
+        )
+        info = torchaudio.info(wav)
+
+        resampled = sig
+        # Maybe resample to 16kHz
+        if int(info.sample_rate) != int(hparams["sample_rate"]):
+            resampled = torchaudio.transforms.Resample(
+                info.sample_rate, hparams["sample_rate"],
+            )(sig)
+
+        resampled = resampled.transpose(0, 1).squeeze(1)
+        # Select the proper audio channel of the segment
+        if channel == "A":
+            resampled = resampled[:, 0]
+        else:
+            resampled = resampled[:, 1]
+
+        if hparams["speed_perturb"]:
+            # sig = sb.dataio.dataio.read_audio(wav)
+            # factor = np.random.uniform(0.95, 1.05)
+            # sig = resample(sig.numpy(), 16000, int(16000*factor))
+            speed = sb.processing.speech_augmentation.SpeedPerturb(
+                16000, [x for x in range(95, 105)]
+            )
+            resampled = speed(resampled.unsqueeze(0)).squeeze(
+                0
+            )  # torch.from_numpy(sig)
+        return resampled
+
+    sb.dataio.dataset.add_dynamic_item([train_data], audio_pipeline_train)
+
+    # 3. Define text pipeline:
+    @sb.utils.data_pipeline.takes("words")
+    @sb.utils.data_pipeline.provides(
+        "words", "tokens_list", "tokens_bos", "tokens_eos", "tokens"
+    )
+    def text_pipeline(words):
+        yield words
+        tokens_list = tokenizer.encode_as_ids(words)
+        yield tokens_list
+        tokens_bos = torch.LongTensor([hparams["bos_index"]] + (tokens_list))
+        yield tokens_bos
+        tokens_eos = torch.LongTensor(tokens_list + [hparams["eos_index"]])
+        yield tokens_eos
+        tokens = torch.LongTensor(tokens_list)
+        yield tokens
+
+    sb.dataio.dataset.add_dynamic_item(datasets, text_pipeline)
+
+    # 4. Set output:
+    sb.dataio.dataset.set_output_keys(
+        datasets, ["id", "sig", "words", "tokens_bos", "tokens_eos", "tokens"],
+    )
+
+    return (
+        train_data,
+        valid_data,
+        test_datasets,
+        tokenizer,
+    )
+
+
+if __name__ == "__main__":
+    # CLI:
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # If distributed_launch=True then
+    # create ddp_group with the right communication protocol
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    # 1.  # Dataset prep (parsing Librispeech)
+    from switchboard_prepare import prepare_switchboard  # noqa
+    from normalize_util import normalize_words, read_glm_csv  # noqa
+
+    normalize_fn = functools.partial(
```

Comment:
`normalize_fn` needs to be initialized AFTER `prepare_switchboard` is executed or you'll get a `FileNotFoundError` because the glm file won't be available yet.

Reply:
Fixed in 34c0f62aa49b08155766997b443de58c13dcf399.